### ISSUE TITLE
When story id starts with 'settings' the addons panel is not responding 🤯 

### ISSUE THREAD

**Issue by darkowic** [storybookjs/storybook#14718]
**Describe the bug**
When story id starts with `settings`, the addon panel is frozen/broken/not responding. 🤯 

https://user-images.githubusercontent.com/11510581/116053981-83f51f80-a67b-11eb-8a70-28e8b989b50e.mp4

**To Reproduce**
I reproduced it on newly created CRA with SB. See video above.

To reproduce: just add story `id` starting with `settings`

```jsx
export default {
  title: "Example/Button",
  id: "settings-card",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
  },
};
```

**System**

```
Environment Info:

  System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 14.14.0 - ~/.nvm/versions/node/v14.14.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v14.14.0/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v14.14.0/bin/npm
  Browsers:
    Chrome: 90.0.4430.85
    Firefox: 85.0.1
    Safari: 14.0.3
  npmPackages:
    @storybook/addon-actions: ^6.2.9 => 6.2.9
    @storybook/addon-essentials: ^6.2.9 => 6.2.9
    @storybook/addon-links: ^6.2.9 => 6.2.9
    @storybook/node-logger: ^6.2.9 => 6.2.9
    @storybook/preset-create-react-app: ^3.1.7 => 3.1.7
    @storybook/react: ^6.2.9 => 6.2.9
```

**Comment by shilman** [comment]
Sounds like a routing bug, since we have pages like `/settings/about` and `/settings/shortcuts`. Anybody want to take a crack at this?

**Comment by jay24rajput** [comment]
Hey! I am new to the codebase still exploring the functionality! I would like to work on this issue as my first issue on storybook. Any getting started or tricks and tips?

**Comment by shilman** [comment]
@jay24rajput thank you! 🙏 
 check out https://github.com/storybookjs/storybook/blob/next/CONTRIBUTING.md#development-guide to get started 💪

**Comment by rsotoMrOrange** [comment]
Hi! I would like to try to fix this issue, but I see someone else already started. Can I give this one a go or should I look for another one?

**Comment by darleendenno** [comment]
‼️ This issue is available for the taking! ‼️ 

@rsotoMrOrange? @jay24rajput? are either of you interested?

**Comment by vivek12345** [comment]
I would like to try working on this if none of the others are interested

**Comment by darleendenno** [comment]
> I would like to try working on this if none of the others are interested

Go for it! 😄

**Comment by vivek12345** [comment]
@darleendenno would you suggest a particular file to start with?

**Comment by vivek12345** [comment]
actually looking at this PR https://github.com/storybookjs/storybook/pull/14822/
I think most of the work is already done. May be I will have to add few test cases and update useMemo dependency array

**Comment by ben-m-j-taylor** [comment]
@darleendenno Hi, I submitted a [PR](https://github.com/storybookjs/storybook/pull/15910) for this issue last night, I hope that was ok? I probably should have checked first, but it didn't look like there had been too much activity on the ticket for the last few weeks.

**Comment by pahan35** [comment]
@shilman what is the correct link to settings?

**Comment by shilman** [comment]
@pahan35 

https://next--storybookjs.netlify.app/official-storybook/?path=/settings/shortcuts
https://next--storybookjs.netlify.app/official-storybook/?path=/settings/about

**Comment by pahan35** [comment]
I guess, this issue should be fixed after merging my PR?
cc: @shilman

**Comment by DaveLak** [comment]
@shilman @ndelangen Correct me if I'm wrong but it looks like this issue can be closed as complete. @pahan35's fix from #16963 has been incorporated in the stable releases [since at least v6.5.0](https://github.com/storybookjs/storybook/blob/05070ca621fcebd46cd6c58fcb0eb24a35bbe651/lib/ui/src/app.tsx#L51).
